crates/libwebrtc-sys/build.rs

```diff
@@ -501,83 +494,350 @@ fn compile_openal() -> anyhow::Result<()> {
+            temp_dir,
+            checksum,
+            artifact: self,
+        }))
+    }
+
+    /// Get name of the libwebrtc archive.
+    fn archive_name() -> anyhow::Result<String> {
         let mut name = String::from("libwebrtc-");
 
         #[cfg(target_os = "windows")]
-        name.push_str("windows-x64.tar.gz");
+        name.push_str("windows-x64");
         #[cfg(target_os = "linux")]
-        name.push_str("linux-x64.tar.gz");
+        name.push_str("linux-x64");
 
         match get_target()?.as_str() {
             "aarch64-apple-darwin" => {
-                name.push_str("macos-arm64.tar.gz");
+                name.push_str("macos-arm64");
             }
             "x86_64-apple-darwin" => {
-                name.push_str("macos-x64.tar.gz");
+                name.push_str("macos-x64");
             }
             _ => (),
         }
 
-        name
-    };
-    let archive = temp_dir.join(&tar_file);
-    let checksum = lib_dir.join("CHECKSUM");
-    let expected_hash = get_expected_libwebrtc_hash()?;
-
-    // Force download if `INSTALL_WEBRTC=1`.
-    if env::var("INSTALL_WEBRTC").as_deref().unwrap_or("0") == "0" {
-        // Skip download if already downloaded and checksum matches.
-        if fs::metadata(&lib_dir).is_ok_and(|m| m.is_dir())
-            && fs::read(&checksum).unwrap_or_default().as_slice()
-                == expected_hash.as_bytes()
-        {
-            return Ok(());
-        }
+        name.push_str(".tar.gz");
+
+        Ok(name)
     }
+}
 
-    // Clean up `temp` directory.
-    if temp_dir.exists() {
-        fs::remove_dir_all(&temp_dir)?;
+/// Representation of an artifact from GitHub API.
+#[derive(Deserialize)]
+struct ArtifactMetadata {
+    /// Hash of artifact's archive content.
+    digest: Cow<'static, str>,
+    /// Url to REST API for getting artifact's download link.
+    archive_download_url: String,
+}
+
+/// Response from list artifacts [endpoint][1] of GitHub API.
+///
+/// [1]: https://docs.github.com/en/rest/actions/artifacts
+#[derive(Deserialize)]
+struct ArtifactsResponse {
+    /// List of artifacts metadata.
+    artifacts: Vec<ArtifactMetadata>,
+}
+
+/// Representation of a workflow run from GitHub API.
+#[derive(Deserialize)]
+struct WorkflowRun {
+    /// Url of REST API for getting list of artifacts.
+    artifacts_url: String,
+}
+
+/// Response from list workflow runs [endpoint][1] of GitHub API.
+///
+/// [1]: https://docs.github.com/en/rest/actions/workflow-runs
+#[derive(Deserialize)]
+struct WorkflowRunsResponse {
+    /// List of workflow runs.
+    workflow_runs: Vec<WorkflowRun>,
+}
+
+/// Representation of GitHub repository with build artifacts.
+enum WebrtcRepository {
+    /// Release representation.
+    Release,
+    /// Branch representation.
+    Branch {
+        /// Name of the branch.
+        name: String,
+        /// GitHub token to download the archive.
+        github_token: String,
```

Comment:
Well, I'm probably a bit too paranoid, but let's not store token in structs. Someday, someone will be debugging why it fails, adds a `#[derive(Debug)]` and prints it on CI. Lets either store it as a secret string, or not store at all.

Reply:
@turtletongue ,

+1 for [`secrecy`](https://crates.io/crates/secrecy) usage.